packages/react-ui-components/src/TextInput/textInput.tsx

```diff
@@ -52,11 +52,16 @@ export interface TextInputProps extends Omit<React.InputHTMLAttributes<HTMLInput
 
 class TextInput extends PureComponent<TextInputProps> {
     // tslint:disable-next-line:readonly-keyword
-    private ref?: HTMLInputElement;
+    private ref?: React.RefObject<HTMLInputElement>;
+
+    constructor(props: any) {
+        super(props);
+        this.ref = React.createRef();
+    }
 
     public readonly componentDidMount = (): void => {
-        if (this.ref && this.props.setFocus) {
-            this.ref.focus();
+        if (this.ref && this.ref.current && (this.props.setFocus || this.props.autoFocus)) {
```

Comment:
Where is `autoFocus` prop coming from? It's not defined in proptypes. How does it differ from `setFocus`?

Reply:
The NodeCreationDialog sets the property since some years ;)

Reply:
TextInput also has the prop for some time.

Reply:
Ah, you mean to say that it's is just passed down to the DOM element as a DOM attribute here? https://github.com/neos/neos-ui/blob/72caa0aa8a97ed79ab37a9c701136e8b6405f125/packages/react-ui-components/src/TextInput/textInput.tsx#L107
Didn't think about it at first, good point then.